server/utils.go

```diff
@@ -174,3 +185,196 @@ func (w *withCode) Format(s fmt.State, verb rune) {
+			}
+
+			resourceName := apiResource.Name
+
+			actionId = path.Join(actionId, resourceName)
+
+			for _, operation := range operationsList {
+				if strings.Contains(operation.Name, actionId) {
+					opNameArr := strings.Split(operation.Name, "/")
+
+					if group != "v1" {
+						// extra validation to make sure groups are the same
+						if !(group == opNameArr[2]) {
+							continue
+						}
+					} else {
+						// make sure resources are the same for core apigroup
+						if !(resourceName == opNameArr[2]) {
+							continue
+						}
+					}
+
+					verb := opNameArr[len(opNameArr)-1]
+					if verb == "action" {
+						verb = path.Join(opNameArr[len(opNameArr)-2], opNameArr[len(opNameArr)-1])
+					}
+
+					da := oputil.DataAction{
+						ActionInfo: oputil.AuthorizationActionInfo{
+							IsDataAction: true,
+						},
+						IsNamespacedResource: apiResource.Namespaced,
+					}
+					da.ActionInfo.AuthorizationEntity.Id = operation.Name
+
+					if operationsMap[group] == nil {
+						operationsMap[group] = []map[string]map[string]oputil.DataAction{}
+					}
+					resourceAndVerb := map[string]map[string]oputil.DataAction{}
+					resourceAndVerb[resourceName] = map[string]oputil.DataAction{}
+					resourceAndVerb[resourceName][verb] = da
+					operationsMap[group] = append(operationsMap[group], resourceAndVerb)
+				}
+			}
+		}
+	}
+
+	klog.V(5).Infof("Operations list: %v", operationsMap)
+
+	return operationsMap, nil
+}
+
+func fetchApiResources() ([]*metav1.APIResourceList, error) {
+	// creates the in-cluster config
+	klog.V(5).Infof("Fetch apiresources list")
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "Error building kubeconfig")
+	}
+
+	kubeclientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error building kubernetes clientset")
+	}
+
+	apiresourcesList, err := kubeclientset.Discovery().ServerPreferredResources()
+	if err != nil {
+		return nil, err
+	}
+
+	klog.V(5).Infof("List of ApiResources fetched from apiserver: %v", apiresourcesList)
+
+	return apiresourcesList, nil
+}
+
+func fetchDataActionsList(environment string, clusterType string, loginURL string, tenantID string) ([]oputil.Operation, error) {
+	env := azure.PublicCloud
+	var err error
+	if environment != "" {
+		env, err = azure.EnvironmentFromName(environment)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to parse environment for Azure.")
+		}
+	}
+
+	endpoint := ""
+
+	switch clusterType {
+	case oputil.ConnectedClusters:
+		endpoint = fmt.Sprintf(oputil.OperationsEndpointFormatARC, env.ResourceManagerEndpoint)
+	case oputil.ManagedClusters:
+		endpoint = fmt.Sprintf(oputil.OperationsEndpointFormatAKS, env.ResourceManagerEndpoint)
+	case oputil.Fleets:
+		endpoint = fmt.Sprintf(oputil.OperationsEndpointFormatAKS, env.ResourceManagerEndpoint)
+	default:
+		return nil, errors.New("Failed to create endpoint for Get Operations call.")
```

Comment:
nit: should be unsupported clusterType: %s

Reply:
done